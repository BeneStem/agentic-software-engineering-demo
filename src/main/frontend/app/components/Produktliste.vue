<template>
  <div class="erk_finden__produktliste">
    <Produktkachel
      v-for="(produkt, index) in produkte"
      :key="produkt.nummer"
      :produkt="produkt"
      :position="index"
      :listenname="trackingname"
      :listennummer="trackingnummer"
    />
  </div>
</template>

<script setup lang="ts">
import { Produkt } from '../model/Produkt';
import Produktkachel from '@/main/frontend/app/components/Produktkachel.vue';

interface ProduktlisteProps {
  produkte: Produkt[];
  trackingname: string;
  trackingnummer: number;
}

defineProps<ProduktlisteProps>();
</script>

<style scoped lang="scss">
.erk_finden__produktliste {
  display: grid;
  grid-template-columns: 1fr;
  grid-column-gap: 40px;
  grid-row-gap: 20px;
}

@media (min-width: $breakpoint-small) {
  .erk_finden__produktliste {
    grid-template-columns: 1fr 1fr;
  }
}

@media (min-width: $breakpoint-large) {
  .erk_finden__produktliste {
    grid-template-columns: 1fr 1fr 1fr;
  }
}
</style>
