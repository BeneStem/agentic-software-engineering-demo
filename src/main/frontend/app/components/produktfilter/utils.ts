import { FilterTyp } from '@/main/frontend/app/model/FilterTyp';

export function erstelleFilterTabId(filterTyp: FilterTyp): `filter-tab-${FilterTyp}` {
  return `filter-tab-${filterTyp}`;
}

export function erstelleFilterTabPanelId(filterTyp: FilterTyp): `filter-tabpanel-${FilterTyp}` {
  return `filter-tabpanel-${filterTyp}`;
}
