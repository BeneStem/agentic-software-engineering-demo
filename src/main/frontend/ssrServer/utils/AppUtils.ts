import serializeJavascript from 'serialize-javascript';

export function renderAppState(appId: string, storeKey: string, state: unknown): string {
  const localStoreKey = `${storeKey}__${appId}`;
  const autoRemove =
    ';(function(){var s;(s=document.currentScript||document.scripts[document.scripts.length-1]).parentNode.removeChild(s);}());';
  const initializeStateScript = `
    <script type='text/javascript'>
      if(!window.__INITIAL_STATE__) {
          window.__INITIAL_STATE__ = {}
      }
      window.__INITIAL_STATE__["${localStoreKey}"] = ${serializeJavascript(state)}
      ${autoRemove}
    </script>
  `;
  return state ? initializeStateScript : '';
}
