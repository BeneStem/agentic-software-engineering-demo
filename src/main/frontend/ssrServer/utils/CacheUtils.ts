const oneSecond = 1;
const oneMinute = 60 * oneSecond;
const oneHour = 60 * oneMinute;
const oneDay = 24 * oneHour;
const oneYear = 365 * oneDay;

const cacheTimeInSeconds = {
  staticFiles: oneYear * 30,
  services: oneSecond * 10,
};

export const buildCacheControlHeaderValue = (cacheTimeInSeconds: number): string => {
  if (cacheTimeInSeconds > 0) {
    return `public, max-age=${cacheTimeInSeconds}`;
  }
  return 'no-cache, no-store';
};

export const getCacheControlHeaderValueForStaticFiles = (): string => {
  return buildCacheControlHeaderValue(cacheTimeInSeconds.staticFiles);
};

export const getCacheControlHeaderValueForServices = (): string => {
  return buildCacheControlHeaderValue(cacheTimeInSeconds.services);
};
