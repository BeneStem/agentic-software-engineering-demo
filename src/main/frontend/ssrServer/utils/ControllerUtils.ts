export interface TagOptions {
  appName: string;
  filePath?: string;
}

export function createCssLinkTag(options: TagOptions): string {
  const { appName, filePath } = options;
  return filePath ? `<link rel='stylesheet' href='/finden/static/${appName}/client${filePath}' />` : '';
}

export function createScriptTag(options: TagOptions): string {
  const { appName, filePath } = options;
  return filePath ? `<script type='text/javascript' src='/finden/static/${appName}/client${filePath}'></script>` : '';
}
