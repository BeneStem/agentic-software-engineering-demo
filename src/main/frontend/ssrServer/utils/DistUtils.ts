import path from 'path';

const DEFAULT_RELATIVE_DIST_PATH = '../../../dist';
const TEST_RELATIVE_DIST_PATH = '../../../../test/frontend/ssrServer/testDist';

const RELATIVE_DIST_PATH = process.env.NODE_ENV === 'test' ? TEST_RELATIVE_DIST_PATH : DEFAULT_RELATIVE_DIST_PATH;

export const DIST_PATH = path.join(__dirname, RELATIVE_DIST_PATH);
