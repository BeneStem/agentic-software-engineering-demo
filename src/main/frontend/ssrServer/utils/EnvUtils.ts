import type { FastifyInstance } from 'fastify';
import path from 'path';
import { DIST_PATH } from './DistUtils';
import fastifyEnv from '@fastify/env';

export const FASTIFY_SERVER_DEFAULT_PORT = 8080;

export async function registerEnvironment(fastifyInstance: FastifyInstance): Promise<void> {
  const DOT_ENV_PATH = path.join(DIST_PATH, '../.env.fastify');
  const schema = {
    type: 'object',
    required: ['FASTIFY_SERVER_PORT'],
    properties: {
      FASTIFY_SERVER_PORT: {
        type: 'number',
        default: FASTIFY_SERVER_DEFAULT_PORT,
      },
      IS_LOCAL_DEPLOYMENT: {
        type: 'string',
        default: 'false',
      },
      DEBUG_MEMORY_USAGE: {
        type: 'string',
        default: 'false',
      },
    },
  };
  await fastifyInstance.register(fastifyEnv, {
    schema,
    dotenv: {
      path: DOT_ENV_PATH,
    },
  });
}
