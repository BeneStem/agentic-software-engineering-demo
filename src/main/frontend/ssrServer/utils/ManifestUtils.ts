import path from 'path';
import { AppManifest, AppType, RenderSsrApp } from '../types';
import { DIST_PATH } from './DistUtils';

export function loadClientManifest(appName: string): AppManifest {
  return loadManifest(appName, 'client');
}

export function loadSsrManifest(appName: string): AppManifest {
  return loadManifest(appName, 'ssr');
}

export function loadRenderSsrApp(appName: string, ssrManifest: AppManifest): RenderSsrApp {
  return require(getSsrAppPath(appName, ssrManifest)).default;
}

function loadManifest(appName: string, appType: AppType): AppManifest {
  return require(path.join(DIST_PATH, `/${appName}/${appType}/${appType}-manifest.json`));
}

function getSsrAppPath(appName: string, manifest: AppManifest): string {
  const APP_DIST_PATH = path.join(DIST_PATH, appName);
  return path.join(APP_DIST_PATH, '/ssr', manifest['app.js']);
}
