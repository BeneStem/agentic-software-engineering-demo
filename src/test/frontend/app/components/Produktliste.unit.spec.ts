import { shallowMount } from '@vue/test-utils';
import Produktliste from '@/main/frontend/app/components/Produktliste.vue';
import { testProduktListe } from '@/test/frontend/utils/TestProduktHelper';

describe('Produktliste.vue', () => {
  it('Rendert eine Produktliste', () => {
    // Given
    const wrapper = shallowMount(Produktliste, {
      props: {
        produkte: testProduktListe,
        trackingname: 'trackingname',
        trackingnummer: 1
      }
    });

    // Then
    expect(wrapper.element).toMatchSnapshot();
  });
});
