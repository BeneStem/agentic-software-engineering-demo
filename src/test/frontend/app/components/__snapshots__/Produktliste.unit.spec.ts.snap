// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Produktliste.vue Rendert eine Produktliste 1`] = `
<div
  class="erk_finden__produktliste"
>
  
  <produktkachel-stub
    listenname="trackingname"
    listennummer="1"
    position="0"
    produkt="[object Object]"
  />
  <produktkachel-stub
    listenname="trackingname"
    listennummer="1"
    position="1"
    produkt="[object Object]"
  />
  
</div>
`;
